profile-widget/diveprofileitem.cpp

```diff
@@ -163,37 +168,93 @@ void DiveProfileItem::replot(const dive *d, int from, int to, bool in_planner)
 	pat.setColorAt(0, getColor(DEPTH_TOP));
 	setBrush(QBrush(pat));
 
-	int last = -1;
-	for (int i = from; i < to; i++) {
-		struct plot_data *pd = dataModel.data().entry;
-		struct plot_data *entry =  pd + i;
-		// "min/max" are the 9-minute window min/max indices
-		struct plot_data *min_entry = pd + entry->min;
-		struct plot_data *max_entry = pd + entry->max;
-
-		if (entry->depth < 2000)
-			continue;
+	// No point in searching peaks with less than three samples
+	if (to - from < 3)
+		return;
 
-		if ((entry == max_entry) && entry->depth / 100 != last) {
-			plot_depth_sample(entry, Qt::AlignHCenter | Qt::AlignBottom, getColor(SAMPLE_DEEP));
-			last = entry->depth / 100;
+	const int half_interval = vAxis.getMinLabelDistance(hAxis);
+	const int min_depth = 2000; // in mm
+	const int min_prominence = 2000; // in mm (should this adapt to depth range?)
```

Comment:
hard to tell - and likely a matter of taste... for now I think this is a reasonable choice